apps/passport-server/src/services/telegramService.ts

```diff
@@ -51,17 +62,44 @@ export class TelegramService {
     this.context = context;
     this.rollbarService = rollbarService;
     this.bot = bot;
+    this.fetchLinkMenu = async (): Promise<MenuRange<Context>> =>
+      new MenuRange();
 
     this.bot.api.setMyDescription(
-      "I'm the Research Workshop ZK bot! I'm managing the Research Workshop Telegram group with ZKPs. Press START to get started!"
+      "I'm the ZK Auth Bot! I'm managing fun events with ZKPs. Press START to get started!"
     );
 
-    this.bot.api.setMyShortDescription(
-      "Research Workshop ZK Bot manages the Research Workshop Telegram group using ZKPs"
-    );
+    this.bot.api.setMyShortDescription("ZK Auth Bot manages events using ZKPs");
+
+    const pcdPassMenu = new Menu("pcdpass");
+    const eventsMenu = new Menu("events");
+
+    // Uses the dynamic range feature of Grammy menus https://grammy.dev/plugins/menu#dynamic-ranges
+    // This callback function is inline due to the context.dbPool not being defined when putting this logic in a member function
+    // This function needs access to the global state (this.events and this.deleteEvents).
+    // TODO: Figure out a better way to execute the logic with the required state.
+    eventsMenu.dynamic(async () => {
+      logger(`[TELEGRAM] calling dynamic events menu...`);
+      return await this.fetchLinkMenu();
+    });
+
+    pcdPassMenu.dynamic((ctx, range) => {
+      const userId = ctx?.from?.id;
+      const username = ctx?.from?.username;
+      const firstName = ctx?.from?.first_name;
+      const name = firstName || username;
+
+      if (userId && name) {
+        const proofUrl = this.generateProofUrl(userId.toString());
+        range.webApp(`Generate ZKP for ${name} 🚀`, proofUrl);
+      } else {
+        ctx.reply(`userId or name not found...`);
```

Comment:
Let's make this reply a bit more human readable for a nontechnical attendee

Reply:
Done: `Unable to locate your Telegram account. Please try again, or contact passport@0xparc.org`